src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -745,10 +761,48 @@ private static CipherSuite chooseCipherSuite(
             }
 
             // no cipher suites in common
+            printServerEnabledCipherSuites(shc, null, null);
             return null;
         }
     }
 
+    /* *
+     *  When debugging enabled with the value of "ssl, handshake", print out enabled cipher suites on the server side
+     */
+    private static void printServerEnabledCipherSuites(sun.security.ssl.ServerHandshakeContext shc,
```

Comment:
maybe use `printServerSocketConfig` might suit better for method name ?

Reply:
Updated the code, see new commit.